### PR TITLE
Schema Update and cleanword default

### PR DESCRIPTION
**Pull request by c-o-m-m-a-n-d-e-r**
added cleanword default

removed rooms from schema for ui (not nessesary at this moment)

added autorooms for getting room ids into Log

### CODE REVIEW

**Comment by c-o-m-m-a-n-d-e-r** [comment]
I think please assign 
https://github.com/nicoh88/homebridge-xiaomi-roborock-vacuum/issues/115

File: index.js [hunk 1]
```diff
@@ -552,9 +553,9 @@ class XiaomiRoborockVacuum {
     try {
       const map = await this.device.call('get_room_mapping');
       this.log.info(`INF getRoomMap | ${this.model} | Map is ${map}`);
```
Comment (afharo) [inline index.js]: Actually, when are we calling the `getRoomMap()` method?
It's never called. The new options defined are never used xD

File: config.schema.json [hunk 2]
```diff
@@ -45,21 +45,11 @@
         "default": "clean",
         "required": true
       },
-      "rooms" {
```
Comment (afharo) [inline config.schema.json]: I don't think removing them are OK. You can see there are some users setting up the plugin via the UI.

Comment (c-o-m-m-a-n-d-e-r) [reply config.schema.json]: ok, can you please help me with the schema then?

Comment (afharo) [reply config.schema.json]: isn't it just adding the missing `:` and removing the `default` entry? It's an optional parameter, if not defined, that's fine

File: index.js (hunk 1 above)
Comment (c-o-m-m-a-n-d-e-r) [reply index.js]: sure, it is cvalled if you add the config : "autoroom".

it is not in the readme yes ... it was a first try and many people don't know how to get the IDs.
so i think will be good to let them take it out of the log.

i see a good point here https://github.com/nicoh88/homebridge-xiaomi-roborock-vacuum/issues/114

i changed the code so its called when configured AND device is discovered

File: config.schema.json (hunk 2 above)
Comment (c-o-m-m-a-n-d-e-r) [reply config.schema.json]: no ... please try yourself with this in it:
`"rooms": {
        "title": "Room cleaning",
        "type": "array",
        "required": false,
        "items": {
          "name": {
            "name": "Room Name",
            "type": "string"
          },
          "id": {
            "name": "Room ID",
            "type": "integer"
          }
        }
      }`

and you will see the next error

Comment (afharo) [reply config.schema.json]: Yeah... missing the`properties` attribute :)
```
    "rooms": {
        "title": "Room cleaning",
        "type": "array",
        "required": false,
        "items": {
          "type": "object",
          "properties": {
            "name": {
              "name": "Room Name",
              "type": "string"
            },
            "id": {
              "name": "Room ID",
              "type": "integer"
            }
          }
        }
      }
```

File: index.js (hunk 1 above)
Comment (afharo) [reply index.js]: > sure, it is cvalled if you add the config : "autoroom".

Can you point me at the line of code in index.js where you check this config and call the method? I can't seem to find it... 🤔

Comment (afharo) [reply index.js]: Oh! just saw your latest commit 👍

File: config.schema.json (hunk 2 above)
Comment (c-o-m-m-a-n-d-e-r) [reply config.schema.json]: oh great ... will update, gimme a minute

Comment (c-o-m-m-a-n-d-e-r) [reply config.schema.json]: Working! Great. Its in now :-)

File: index.js [hunk 3]
```diff
@@ -347,6 +348,10 @@ class XiaomiRoborockVacuum {
       this.log.info('STA getDevice | FanSpeed: ' + this.device.property("fanSpeed"));
       this.log.info('STA getDevice | BatteryLevel: ' + this.device.property("batteryLevel"));
 
+      if (this.config.autoroom) {
+        this.getRoomMap();
```
Comment (afharo) [inline index.js]: Mind the method is a promise, you need to `await` it or you'll get an 'unhandled promise exception'.
```suggestion
        await this.getRoomMap();
```

Comment (c-o-m-m-a-n-d-e-r) [reply index.js]: yes you are totally right ... sorry i'm on medicin at the moment ... sleep the hole day ...

**Comment by afharo** [comment]
@nicoh88

**Comment by c-o-m-m-a-n-d-e-r** [comment]
I updated the UI Schema a little bit so it looks better now.
IP and Token in one line and the Room Section looks pretty good now.

Bildschirmfoto 2020-02-27 um 08 47 46

**Comment by nicoh88** [comment]
Thanks to @c-o-m-m-a-n-d-e-r and @afharo

**Pull request merged**